msal/src/main/java/com/microsoft/identity/client/PublicClientApplication.java

```diff
@@ -1512,22 +1513,22 @@ public void onCancel() {
             final CommandResult commandResult = future.get();
```

Comment:
We should think about having a timeout for the  future.get() call, so it doesn't wait indefinitely..